rumqttd/src/router/routing.rs

```diff
@@ -1129,15 +1232,52 @@ fn forward_device_data(
         return ConsumeStatus::FilterCaughtup;
     }
 
+    // there are three cases
+    // 1. we are using an alias for filter already
+    // 2. we can't use alias as none is avaliable to use
+    // 3. we need to set the alias as this is first time
+    let mut clear_topic = true;
```

Comment:
we clear topic when we want to use the topic alias, so it would be opposite, more like `topic_alias_exists`, but as we use it to decide whether or not to clear topic, it is named `clear_topic` ( after struggling for so long to come up with good name )